### PR TITLE
Fix stacked commands

### LINKED ISSUE #134: User can still execute other commands while awaiting confirmation for a previous command
**Describe the bug**
User can still execute other commands while awaiting confirmation for a previous command.

e.g a user can key in the following commands (in order)
```
delete 1
delete 1
delete 1
y
y
y
```
and this will delete the first 3 commands

While this may not be a big issue, there can also be cases where this happens
```
delete 1
add n/New Person p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney
y
```
In this case, the behavior is indeterminate. If the new person is sorted behind the original then the original will still be deleted, but if the former is in front then it be deleted instead.

### PR DESCRIPTION
**Pull request by mongj**
Resolves #134 

As per #134, I realized using a stack may not be the best abstraction for keeping track of commands to confirm, due to the indeterminate behavior as described in the issue.

I've reverted the implementation to use a single static variable within the `Command` class to keep track of the command to confirm. The expected behavior after this PR is that `y` or `n` will be the only acceptable inputs if the user has previously triggered a command that requires confirmation.

The `CommandStack` is kept there in case we want to implement undo/redo or similar features in the future.

### CODE REVIEW

**Review by SuhailLoya — COMMENTED** [review]
I feel we shouldn't give the responsibility of checking whether there is an unconfirmed command to Command classes to adhere to the single responsibility principle

Perhaps a better place to make this check is at logic manager.

File: src/main/java/careconnect/logic/commands/DeleteLogCommand.java [hunk 1]
```diff
@@ -92,9 +94,9 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         if (this.requireConfirmation) {
-            // Add a DeleteLogCommand to the stack
-            Command.STACK.add(new DeleteLogCommand(personIndex, logIndex, false));
-            return new CommandResult(Command.CONFIRMATION_MESSAGE,
+            // Queues a DeleteLogCommand
+            Command.commandToConfirm = new DeleteLogCommand(personIndex, logIndex, false);
```
Comment (AppleJem) [inline src/main/java/careconnect/logic/commands/DeleteLogCommand.java:98]: This looks repeated in every Command, is it possible to extract this out into another function? Maybe a function in the parent Command class that takes in a Command (in this case DeleteLogCommand)?

Comment (AppleJem) [reply src/main/java/careconnect/logic/commands/DeleteLogCommand.java:98]: Prolly not important tho hahaha

**Review by AppleJem — APPROVED** [review]
Looks gooood! Not sure my suggestion makes sense honestly, lemme know

**Comment by AppleJem** [comment]
> I feel we shouldn't give the responsibility of checking whether there is an unconfirmed command to Command classes to adhere to the single responsibility principle
> 
> Perhaps a better place to make this check is at logic manager.

Oooh I didn't see this. Yeah that makes a lot of sense actually, then can avoid the repeating of the logic in every Command oso

**Comment by mongj** [comment]
> I feel we shouldn't give the responsibility of checking whether there is an unconfirmed command to Command classes to adhere to the single responsibility principle
> 
> Perhaps a better place to make this check is at logic manager.

Thanks for the comment! that's a great suggestion and I have amended my PR accordingly

Just a note, the `commandToConfirm` field and the associated accessors are all at the class-level. The alternative is to somehow inject an instance of `LogicManager` into the command, perhaps in the `execute` method in `LogicManager`?

File: src/test/java/careconnect/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -77,6 +78,20 @@ public void autocompleteCommand_availableOptions_success() throws Exception {
...
+        assertCommandException("n", Messages.MESSAGE_NO_EXECUTABLE_COMMAND);
+    }
+
+    @Test
+    public void commandConfirmation_commandYetToConfirm_throwsCommandException() {
+        LogicManager.setCommandToConfirm(new ClearCommand());
+        assertCommandException("clear", Messages.MESSAGE_UNCOMFIRMED_COMMAND);
+        LogicManager.setCommandToConfirm(null);
+    }
+
```
Comment (SuhailLoya) [inline src/test/java/careconnect/logic/LogicManagerTest.java:94]: cool! thanks for the tests

File: src/main/java/careconnect/logic/LogicManager.java [hunk 3]
```diff
@@ -50,6 +50,8 @@ public class LogicManager implements Logic {
             "Could not save data to file %s due to insufficient permissions to write to the file "
                     + "or the folder.";
 
+    private static Command commandToConfirm = null;
+
```
Comment (SuhailLoya) [inline src/main/java/careconnect/logic/LogicManager.java:54]: Yep, looks good, logic manager seems to be the most appropriate component to place this responsibility! thanks!

**Review by SuhailLoya — APPROVED** [review]
LGTM! Very cool! Thanks for resolving the comments

**Pull request merged**